src/modules/CMakeLists.txt

```diff
@@ -1,7 +1,13 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
-add_compile_options("-Wall;-Wextra;-Wunused;-Werror;-Wformat;-Wformat-security;-Wno-unused-result;-Wunused-const-variable=2")
+add_compile_options("-Wall;-Wextra;-Wunused;-Werror;-Wformat;-Wformat-security;-Wno-unused-result")
+if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
+    add_compile_options("-Wunused-const-variable=2")
+endif()
+if (CMAKE_C_COMPILER_ID STREQUAL "Clang")
+    add_compile_options("-Wunused-variable")
+endif()
 

```

Comment:
Can you try using an 'elseif' here to avoid two separate sets of if and endif?

Like this:

if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
    add_compile_options("-Wunused-const-variable=2")
elseif (CMAKE_C_COMPILER_ID STREQUAL "Clang")
    add_compile_options("-Wunused-variable")
endif()